// react-spotify-api/dist/Artist/ArtistTracks.js
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import PropTypes from 'prop-types';
import ApiRequest from '../ApiRequest/ApiRequest';
var BASE_URL = 'https://api.spotify.com/v1/artists';

function ArtistTracks(props) {
  var url = BASE_URL + "/".concat(props.id, "/top-tracks");

  var options = _objectSpread({}, props.options);

  return React.createElement(ApiRequest, {
    url: url,
    options: options
  }, function (data, loading, error) {
    return props.children(data, loading, error);
  });
}

ArtistTracks.propTypes = {
  options: PropTypes.shape({
    market: PropTypes.string.isRequired
  }),
  id: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired
};
ArtistTracks.defaultProps = {
  options: {
    market: 'from_token'
  }
};
export default ArtistTracks;
